src/py2spack/loading.py

```diff
@@ -0,0 +1,266 @@
+"""Utils for downloading packages from PyPI and opening the pyproject.tomls."""
+
+import io
+import tarfile
+from typing import Any, Dict, List, Optional
+
+import requests  # type: ignore
+import tomli
+from packaging import version as vn
+
+KNOWN_ARCHIVE_FORMATS = [
+    ".tar",
+    ".tar.gz",
+    ".tar.bz2",
+    ".gz",
+    ".xz",
+    ".bz2",
+]
+
+
+class APIError(Exception):
+    """Error during version lookup or querying of the PyPI JSON API."""
+
+    def __init__(
+        self,
+        msg: str,
+    ):
+        """Initialize error."""
+        super().__init__(msg)
```

Comment:
The current `__init__` is completely redundant since it's only calling the `__init__` of the superclass. Additionally, the name of the class is too generic and should be more specific, e.g.   
```suggestion
class PyPIQueryError(Exception):   # Or QueryError, or something along these lines
    """Error during version lookup or querying of the PyPI JSON API."""
    pass
    
```